### PR TITLE
Product page with filter

### PR DESCRIPTION
**Pull request by rdthetha**
Files to check:
SideBar.jsx
filterContext
utility.js

### CODE REVIEW

File: src/components/Landing/LandingContent.jsx [hunk 1]
```diff
@@ -7,60 +7,60 @@ import landingimg from '../../image/shoe-final.png'
...
+        <div className="landing-content">
+            <div className="h3">Welcome to ScroleKart! <br />Best Deals on shoes round the year!</div>
+            <div className="small-text">Exclusive collection</div>
+            <Link to='/ProductPage'><button className="btn-primary btn-outline-primary">Explore
                     Now!</button></Link>
             
         </div>
-        <div class="landing-image">
+        <div className="landing-image">
             <img src={landingimg} alt="" />
```
Comment (sri-rishi) [inline src/components/Landing/LandingContent.jsx]: Suggestion: provide `alt` tag a value

File: src/components/Sidebar/Sidebar.jsx [hunk 2]
```diff
@@ -41,16 +51,18 @@ return (
...
+                type="checkbox" checked={showCategory.some((item)=>item==="RunningShoes")}/>Running Shoes</label>
+            <label className='category-label'><input value="CricketShoes" onChange={(event)=>
+                dispatch(categoryHandler(event))}
+                type="checkbox" checked={showCategory.some((item)=>item==="CricketShoes")}/>Cricket Shoes</label>
+            <label className='category-label'><input value="IndoorShoes" onChange={(event)=>
+                dispatch(categoryHandler(event))}
+                type="checkbox" checked={showCategory.some((item)=>item==="IndoorShoes")}/>Indoor Shoes</label>
+            <label className='category-label'><input value="GymShoes" onChange={(event)=>
+                dispatch(categoryHandler(event))}
+                type="checkbox" checked={showCategory.some((item)=>item==="GymShoes")}/>Gym Shoes</label>
```
Comment (sri-rishi) [inline src/components/Sidebar/Sidebar.jsx]: Suggestion: you can make a separate function for checked and pass category value through it and use it here

File: src/pages/Home/LandingPage.jsx [hunk 3]
```diff
@@ -8,9 +8,9 @@ export default function LandingPage(){
 return(
 <>
     <div>
-        <div class="ec-first-container">
+        <div className="ec-first-container">
             <NavigationTop/>
-            <div class="button-category">
+            <div className="button-category">
             <Link to='/ProductPage'><button
                         class="btn-category btn-secondary btn-outline-secondary">Running</button></Link>
```
Comment (sri-rishi) [inline src/pages/Home/LandingPage.jsx]: Suggestion: I think you forgot to change `class` to `className` here

File: src/utility.js [hunk 4]
```diff
@@ -1,9 +1,10 @@
   function getSortedData(productList, sortBy) {
+    let temp=productList;
```
Comment (sri-rishi) [inline src/utility.js:2]: Suggestion: you can use `const` here instead of `let `

**Review by sri-rishi — APPROVED** [review]
Great code , just some suggestion which I think can be, else feel free to merge

File: src/components/Sidebar/Sidebar.jsx [hunk 5]
```diff
@@ -1,15 +1,29 @@
...
+}
 return (
 <div className="ec-side-navbar">
     <div className="ec-product-filter">
-
         <div className="ec-first-element-side-navbar">
             <div className="h3">Filters</div>
-            <a className="h5 italic-text" href="">Clear!</a>
+            <Link to='/ProductPage' onClick={()=> {
+            dispatch({ type: "FILTER_CLEAR"})}}>Clear</Link>
```
Comment (nikhiljugale007) [inline src/components/Sidebar/Sidebar.jsx:26]: suggestion: try to create a separate function for handling events like onchange and onclick

**Review by nikhiljugale007 — COMMENTED** [review]
Did not find any issue, 
Good to merge.

**Pull request merged**